### PR TITLE
STL-535: Search with paging when using validElements API

### PR DESCRIPTION
**Pull request by hung-nguyen-hoang**
Jira: STL-535

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1711/ : SUCCESS in 3s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1528/ : SUCCESS in 7m 28s

**Comment by hung-nguyen-hoang** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1712/ : SUCCESS in 3s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1529/ : SUCCESS in 7m 06s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1715/ : SUCCESS in 3s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1532/ : SUCCESS in 7m 14s

File: lib/gooddata/models/metadata/label.rb [hunk 1]
```diff
@@ -44,14 +44,31 @@ def find_element_value(element_id)
...
+
+          if results['validElements']['items'].count < paging_limit
+            results['validElements']['items'] = all_valid_elements
+            break
+          else
+            offset += paging_limit
+          end
         end
       else
+        # This case will support paging by the method which call this method eg: values(...) method
```
Comment (peter-plochan) [inline lib/gooddata/models/metadata/label.rb:71]: So this means that for the `values` call with the paging set to `1_000_000`, the search won't take into account the 10_000 page limit?

E.g. in this call: https://github.com/gooddata/gooddata-ruby/blob/3bb2e7df7b256b4d6cd44472baf657f99a3f64d5/lib/gooddata/models/user_filters/user_filter_builder.rb#L166

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1716/ : SUCCESS in 5s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1533/ : SUCCESS in 7m 46s

File: lib/gooddata/models/metadata/label.rb (hunk 1 above)
Comment (hung-nguyen-hoang) [reply lib/gooddata/models/metadata/label.rb:71]: It's my missing. I done update PR. Thank you.

**Review by peter-plochan — APPROVED** [review]

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-promote-pipeline/198/ : FAILURE in 6m 54s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/715/ : SUCCESS in 6s
- gooddata-ruby-gitops-deploy-pipeline : SKIPPED

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-promote-pipeline/199/ : FAILURE in 7m 16s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/716/ : SUCCESS in 6s
- gooddata-ruby-gitops-deploy-pipeline : SKIPPED

**Pull request merged**